### DISCUSSION TITLE
Unable to Sign Into ArcGIS Assistant PS

### DISCUSSION THREAD

**Discussion by laurahaley** [EsriPS/arcgis-assistant-feedback#46] [Q&A]
Hi,
I have added ArcGIS Assistant as an item to our 10.8.1 Portal. I believe that I have followed the directions to register it correctly. However when I sign in, I get a blank gray screen with a menu that has the Guides, Feedback, and Option to Sign In again.

![image](https://user-images.githubusercontent.com/15268633/164106917-671a962a-49d5-4ee0-aaf7-0140d0aec56f.png)

We are using enterprise login with our Portal.

**Comment by nheminger** [comment]
Hi @laurahaley, that sounds like there may be an error being thrown. Do you see any errors in the browser developer tools console?

**Reply by laurahaley** [reply]
Hi  @nheminger , Yes there are several errors starting with a CORs error.

Access to fetch at 'https://bloomington.in.gov/arcgis/sharing/rest/community/self?f=json&token=xxxx' from origin 'https://assistant.esri-ps.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

Followed by:

- GET https://bloomington.in.gov/arcgis/sharing/rest/community/self?f=json&token=xxxx request.ts:367 net::ERR_FAILED 200

- Error getting User Session (completeOAuth). Error reading property may result from app redirecting before operation can read token hash in url. Error: Could not create account object. This could be due to UserSession.getUser(). TypeError: Failed to fetch
 
Those errors repeat again then it finishes with a 404.
- GET https://assistant.esri-ps.com/locales/en-US/strings.json 404

**Reply by nheminger** [reply]
@laurahaley thank you, those messages are helpful. This appears to a be a configuration problem with https://bloomington.in.gov/arcgis not returning CORS headers on the responses. When I visit https://bloomington.in.gov/arcgis/sharing/rest/community/self?f=json I see that CORS headers are not being returned. For the requests to work cross-domain those should be present. The site https://enable-cors.org/server.html has some good documentation on enabling those to enable this interaction to work.

**Comment by laurahaley** [comment]
Thanks @nheminger <https://github.com/nheminger> ! I will take a look and
let you know the result.

**Comment by signupleigh** [comment]
Hi @nheminger 

We came across the same problem. We had noticed that if the 'Allow Origin' CORS list on Portal is empty this seems to resolve. In our case, it needs to be populated. Updating the web.config file as per suggested on https://enable-cors.org/server.html has somehow break our survey123 app when connect to our enterprise portal. Has anyone else seen this before please?

**Reply by nheminger** [reply]
Hi @signupleigh, thanks. That is expected when there are 'Allow Origin' entries in [Portal](https://enterprise.arcgis.com/en/portal/latest/administer/windows/restrict-cross-domain-requests-to-your-portal.htm). If there are no entries, then Portal allows all cross-site. As soon as an entry is added to that list then it restricts to only those on the list. Did you happen to try adding the ArcGIS Assistant location as an origin in the list? That would I believe make Portal return the CORS header for you.

What is the error you are seeing that break survey123? My guess would be that once you put in the web.config but have an entry in the 'Allow Origin' CORS list in Portal then likely you are getting two headers (duplicate) set, which is not allowed and will break things. To resolve, you should be able to add ArcGIS Assistant as an allowed origin in Portal and that likely will fix the issue without needing to use the web.config file.

**Reply by signupleigh** [reply]
Thanks so much @nheminger, We actually tried that, too. but the same CORS policy error returned. _from origin 'https://assistant.esri-ps.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

(the tool works nicely with our AGOL though.!)

**Comment by MarcBate** [comment]
Allowed Origins gave same error as others reported. I added this to our web.config

...
    <system.webServer>
        <httpProtocol>
            <customHeaders>
                <add name="Access-Control-Allow-Origin" value="*" />
            </customHeaders>
        </httpProtocol>
...
And no more CORS errors but did get this error:

Error getting User Session (completeOAuth). Error reading property may result from app redirecting before operation can read token hash in url. Error: Could not create account object. This could be due to UserSession.getUser(). ArcGISRequestError: HTTP 401: UNKNOWN_ERROR
